.release-notes/4280.md

```diff
@@ -0,0 +1,5 @@
+## Introduction of empty Ranges
+
+Previously, a `Range` was considered infinite if either 1) the step is 0, or any of min, max, or step were NaN, +Inf or -Inf, or 2) if no progress could be made from min to max due to the sign of the step. This classification was a possible source of bugs including "hanging actors" when a `Range` would otherwise be reasonably expected to be **empty** -- producing no values. Through the RFC process the larger Pony community decided of the agreed-upon new mutually exclusive classification for what constitutes finite, infinite, and empty ranges.
+
+We conducted a search of public Pony code and it appears that the vast majority of `Range` usage should not require any changes. The most likely required change is any infinite `Range` producing an infinite iterator of a single value; for this we recommend switching to use of `Iter[A].repeat_value(value)` from the itertools package.
```

Comment:
Can you add examples of code that requires changing?

Reply:
I added an example of how to change the code to accommodate needing a positive and negative infinite range.